### ISSUE TITLE
Select time for task

### ISSUE THREAD

**Issue by spa49** [frappe/erpnext#6663]
according to this issue:

https://discuss.erpnext.com/t/select-time-for-task/16967

The following is raised.

Please alter field "Expected Start Date" and "Expected End Date" to datetime and have field "Expected End Date" calculated by adding "Expected Time (in hours)" to "Expected Start Date" or vice versa by substracting from "Expected End Date"

**Comment by applepipe** [comment]
This could be usefull for Sales Orders too, to manage delivery time on calendar.

**Comment by Bow-N** [comment]
Yes make the 'Delivery Date' on a 'Sales Order' form to be a 'dateandtime' type field which than can appear on the calendar.

**Comment by Devens** [comment]
+1

**Comment by softwareprofs** [comment]
Any progress on this? I really want to start using ERPNext for task too but the default midnight time doesnt make it possible

**Comment by bojkic** [comment]
showing project tasks in calendar would be essential for me too... also I want to connect Events and Projects too, but when I try that (by making Event doctype child for Project doctype) I run into problems. Any suggestion what is the best practice to relate Projects and Events (oneProject2manyEvents relation)?

**Comment by mukeshmv1** [comment]
+1

**Comment by bennals** [comment]
I'm checking out ERPNext after using Odoo for several years and being frustrated by the lack of what seems like basic functionality, like being able to set Datetime for tasks rather than just Date.  This is one of the first things I've checked out in ERPNext and it has the exact same problem and now what seems like a valid development request appears to have stagnated and gone nowhere.  Very disappointing.

**Comment by bennals** [comment]
I'm checking out ERPNext after using Odoo for several years and being frustrated by the lack of what seems like basic functionality, like being able to set Datetime for tasks rather than just Date.  This is one of the first things I've checked out in ERPNext and it has the exact same problem and now what seems like a valid development request appears to have stagnated and gone nowhere.  Very disappointing.

**Comment by meichthys** [comment]
@ankush I think this is due to the fact that Task `exp_start_date` and `exp_end_date` fields are still currently the `Date` type instead of `Datetime`. This was [fixed in the develop branch](https://github.com/frappe/erpnext/blob/develop/erpnext/projects/doctype/task/task.json#L158), but [hasn't been merged into v15](https://github.com/frappe/erpnext/blob/v15.60.1/erpnext/projects/doctype/task/task.json#L158) as far as I can see: https://github.com/frappe/erpnext/pull/41840

It's appears that this fix was overlooked in the most recent releases.

https://github.com/frappe/erpnext/pull/41633 was never backported to v15.
The pull request for the backport (https://github.com/frappe/erpnext/pull/41840) was killed by the stalebot.

**Comment by meichthys** [comment]
@rohitwaghchaure or @Poorvi-R-Bhat  , Sorry to ping you here (I'm not sure who might be the right person to look at this).
It looks like #41633 was never backported to v15.
The pull request for the backport (https://github.com/frappe/erpnext/pull/41840) was killed by the stalebot.
Would it be possible to get this backport re-opened and merged? The functionality of Tasks is severly limited without being able to specify exact times.

**Comment by pedrovde** [comment]
+1

**Comment by yoc-san** [comment]
@ruthra-kumar Could you please see this?
